app/services/degrees/create_from_hesa.rb

```diff
@@ -32,21 +32,25 @@ def call
     attr_reader :hesa_degrees, :trainee
 
     def create_degrees!
-      hesa_degrees.map do |hesa_degree|
-        next unless importable?(hesa_degree)
+      trainee.transaction do
+        trainee.degrees.destroy_all if hesa_degrees.any?
```

Comment:
Can we check somehow whether this will affect the trainee's timeline? Or would it just get swallowed up in the single "Imported from HESA" timeline entry?

Reply:
I think it would leave a trail. The degree operations would need to be done under `Degree.without_auditing`.

Reply:
It'll definitely create audits, but I'm a tiny bit hopeful that they would be hidden from the timeline. We only show one "Imported from HESA" entry:

<img width="314" alt="Screenshot 2022-10-27 at 10 27 48" src="https://user-images.githubusercontent.com/18436946/198247514-0c7a8e09-9e81-4839-abfb-b22ce1d82c8a.png">

(taken from a production trainee imported from HESA with a degree)

Reply:
what is the difference between created/imported in this context?

Reply:
I can disable audits during the degree deletion and creation quite easily if we're happy with that.

Reply:
I think it's supposed to be the difference between when the record was created in HESA and when it was imported into Register, but looks like that might not be working too well!

Reply:
I'm not 100% sure about disabling audits - the only way i identified these 'bad' degrees was noticing that they all had looooads of audit changes, swapping between non_uk and uk..

Reply:
example: https://www.register-trainee-teachers.service.gov.uk/trainees/SWhmtFzHjPnTRH6bUbMnC4FJ/timeline

however, i'm seeing no 'create' audit for the degrees - so i'm wondering whether the delete and create would just.. not show up somehow?

Reply:
I don't see why they wouldn't show up, creating and deleting a degree manually in console produces timeline entries. Which are the same as the actions used in the HESA import service...

<img width="771" alt="Screenshot 2022-10-27 at 12 31 13" src="https://user-images.githubusercontent.com/6339025/198273426-e0dce762-257b-47bb-8a08-199a758abc40.png">

Reply:
so we would end up with repeats for degree deleted, degree created. which is accurate, although maybe not useful? What would the frequency be like for these creates/deletes?

Reply:
Dave and I had a discussion - we already hid timeline entries for events that happen pre-submitting for TRN, which is why the first degree creation audit is hidden. However, these new create/delete audits would happen _after_ submission.

We realise though, that we could hide all HESA degree audits quite easily from the timeline. That way we can keep the audits for future debugging etc.

Reply:
I might be wary of suppressing _all_ of hesa by default. Might be better to scope suppression to specific actions?

Reply:
@edwardhorsford do you mean suppress just 'create' and 'delete' actions for HESA degrees? since the proposed solution is to just delete and create degrees, there'd be no 'update' actions ever.

Reply:
Sorry I think I misread your comment as hiding *all* HESA audits regardless of type. Ignore me.